### PR TITLE
Fix flaky asv append test

### PR DESCRIPTION
**Pull request by grusev**
#### Checklist

<details>
  <summary>
   FIX Flaky asv append test 
  </summary>
  
The append dataframes are now generated in setup_cache() function, Each new append dataframe has now start segment that is after the previous one. 

As the append dfs are now generated only once the totoal time is lower. Each test now runs 3 instead of 2 times and eventually this should flatten the graphics also

NOTE: Due to the fact that setup_cache() now returns result, there is one additional parameter that (that result) that is added to signature of other functions. Only append functions are using it for now.

ASV time testing in more details with examples for ArcticDB: https://github.com/man-group/ArcticDB/wiki/Running-ASV-Benchmarks

raceback (most recent call last):

File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/asv/benchmark.py", line 1293, in main_run_server

main_run(run_args)

File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/asv/benchmark.py", line 1167, in main_run

result = benchmark.do_run()

File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/asv/benchmark.py", line 573, in do_run

return self.run(*self._current_params)

File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/asv/benchmark.py", line 673, in run

min_run_count=self.min_run_count)

File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/asv/benchmark.py", line 726, in benchmark_timing

timing = timer.timeit(number)

File "/opt/python/cp36-cp36m/lib/python3.6/timeit.py", line 178, in timeit

timing = self.inner(it, self.timer)

File "", line 6, in inner

File "/opt/python/cp36-cp36m/lib/python3.6/site-packages/asv/benchmark.py", line 627, in

func = lambda: self.func(*param)

File "/__w/ArcticDB/ArcticDB/python/benchmarks/basic_functions.py", line 309, in time_append_large

self.lib.append(f"sym", self.df_append_large)

File "/__w/ArcticDB/ArcticDB/python/.asv/env/541b6fbd95fdb429e84621bc981fbd19/lib/python3.6/site-packages/arcticdb/version_store/library.py", line 744, in append

validate_index=validate_index,

File "/__w/ArcticDB/ArcticDB/python/.asv/env/541b6fbd95fdb429e84621bc981fbd19/lib/python3.6/site-packages/arcticdb/version_store/_store.py", line 711, in append

symbol, item, norm_meta, udm, write_if_missing, prune_previous_version, validate_index

arcticdb_ext.exceptions.InternalException: E_ASSERTION_FAILURE Can't append dataframe with start index 2023-01-01 00:00:01.0 to existing sequence ending at 2023-01-12 13:46:40.0

asv: benchmark failed (exit status 1)
 
 - [ ] Have you updated the relevant docstrings, documentation and copyright notice?
 - [ ] Is this contribution tested against [all ArcticDB's features](../docs/mkdocs/docs/technical/contributing.md)?
 - [ ] Do all exceptions introduced raise appropriate [error messages](https://docs.arcticdb.io/error_messages/)?
 - [ ] Are API changes highlighted in the PR description?
 - [ ] Is the PR labelled as enhancement or bug so it appears in autogenerated release notes?
</details>

### CODE REVIEW

**Comment by grusev** [comment]
```
The new version adds logging so that we have more info next time if something fails

· Running 9 total benchmarks (1 commits * 1 environments * 9 benchmarks)

             OUTPUT -------->
             INITIAL DATAFRAME 1000000 rows has Index 2022-12-20 10:13:21 - 2023-01-01 00:00:00
             INITIAL DATAFRAME 1500000 rows has Index 2022-12-14 15:20:01 - 2023-01-01 00:00:00
             Generating dataframe with rows:  1000000
             Generating dataframe no:  0
             STANDARD     Index 2023-01-01 00:00:01 - 2023-01-12 13:46:40
             SHORT_n_WIDE Index 2023-01-01 00:00:01 - 2023-01-01 01:23:20
             Generating dataframe no:  1
             STANDARD     Index 2023-01-24 03:33:21 - 2023-02-04 17:20:00
             SHORT_n_WIDE Index 2023-01-01 02:46:41 - 2023-01-01 04:10:00
             Generating dataframe no:  2
             STANDARD     Index 2023-02-16 07:06:41 - 2023-02-27 20:53:20
             SHORT_n_WIDE Index 2023-01-01 05:33:21 - 2023-01-01 06:56:40
             Generating dataframe no:  3
             STANDARD     Index 2023-03-11 10:40:01 - 2023-03-23 00:26:40
             SHORT_n_WIDE Index 2023-01-01 08:20:01 - 2023-01-01 09:43:20
             Add dataframes:  4
             Generating dataframe with rows:  1500000
             Generating dataframe no:  0
             STANDARD     Index 2023-01-01 00:00:01 - 2023-01-18 08:40:00
             SHORT_n_WIDE Index 2023-01-01 00:00:01 - 2023-01-01 01:23:20
             Generating dataframe no:  1
             STANDARD     Index 2023-02-04 17:20:01 - 2023-02-22 02:00:00
             SHORT_n_WIDE Index 2023-01-01 02:46:41 - 2023-01-01 04:10:00
             Generating dataframe no:  2
             STANDARD     Index 2023-03-11 10:40:01 - 2023-03-28 19:20:00
             SHORT_n_WIDE Index 2023-01-01 05:33:21 - 2023-01-01 06:56:40
             Generating dataframe no:  3
             STANDARD     Index 2023-04-15 04:00:01 - 2023-05-02 12:40:00
             SHORT_n_WIDE Index 2023-01-01 08:20:01 - 2023-01-01 09:43:20
             Add dataframes:  4
             APPEND LARGE cache generation took (s) : 3.471003770828247
                                                                                             ok
[11.11%] ··· basic_functions.ModificationFunctions.time_append_large                                                                    ok
[11.11%] ··· ========= ===========
                rows
             --------- -----------
              1000000   165±300ms
              1500000    205±4ms
             ========= ===========

[22.22%] ··· basic_functions.ModificationFunctions.time_append_short_wide                                                               ok
[22.22%] ··· ========= ============
                rows
             --------- ------------
              1000000   6.25±0.01s
              1500000   6.26±0.02s
             ========= ============

[33.33%] ··· basic_functions.ModificationFunctions.time_append_single                                                                   ok
[33.33%] ··· ========= ============
                rows
             --------- ------------
              1000000   44.7±0.2ms
              1500000   48.9±0.3ms
             ========= ============

[44.44%] ··· basic_functions.ModificationFunctions.time_delete                                                                          ok
[44.44%] ··· ========= ==========
                rows
             --------- ----------
              1000000   53.8±9ms
              1500000   59.1±9ms
             ========= ==========

[55.56%] ··· basic_functions.ModificationFunctions.time_delete_short_wide                                                               ok
[55.56%] ··· ========= ============
                rows
             --------- ------------
              1000000   1.01±0.03s
              1500000   1.01±0.02s
             ========= ============

[66.67%] ··· basic_functions.ModificationFunctions.time_update_half                                                                     ok
[66.67%] ··· ========= ==========
                rows
             --------- ----------
              1000000   122±20ms
              1500000   211±20ms
             ========= ==========

[77.78%] ··· basic_functions.ModificationFunctions.time_update_short_wide                                                               ok
[77.78%] ··· ========= ============
                rows
             --------- ------------
              1000000   6.74±0.02s
              1500000   7.14±0.6s
             ========= ============

[88.89%] ··· basic_functions.ModificationFunctions.time_update_single                                                                   ok
[88.89%] ··· ========= ===========
                rows
             --------- -----------
              1000000   73.1±10ms
              1500000   80.4±20ms
             ========= ===========

[100.00%] ··· basic_functions.ModificationFunctions.time_update_upsert                                                                   ok
[100.00%] ··· ========= ==========
                 rows
              --------- ----------
               1000000   191±30ms
               1500000   280±20ms
              ========= ==========
```

File: python/benchmarks/basic_functions.py
Comment (IvoDD) [inline python/benchmarks/basic_functions.py:352]: We are modifying the `lad` on each run of the benchmark. This relies on subsequent runs of the benchmarks accessing the modified `lad`. I'm not super confident that is the case.

In [here](https://asv.readthedocs.io/en/latest/using.html#environments) I see:
> To improve reproducibility, each benchmark is run in its own process.

File: python/benchmarks/basic_functions.py
Comment (IvoDD) [inline python/benchmarks/basic_functions.py:269]: It looks like you're appending the dataframes in increasing order of timestamps.

But below you're `pop`ing them in decreasing order of timestamps. Don't we want to append them in increasing order?

File: python/benchmarks/finalize_staged_data.py
Comment (IvoDD) [inline python/benchmarks/finalize_staged_data.py:78]: This looks like a temporary fix until #2057 is merged. Can't we wait for that to get merged first and not worry about python 3.6?

**Review by IvoDD — CHANGES_REQUESTED** [review]
Could you write a feature description? I'm interested to know the reasoning behind this change (I know we discussed it offline but it's nice to have it written down).

My understanding is that you've identified that the `teardown` doesn't always execute? Could you provide some evidence that that's the case? It seems like a pretty bad bug in ASV which they should fix if that's the case.

So instead of clearing the library every time in the `teardown` we just append increasing timestamps. If so, shouldn't we 

In fact I think this build passing doesn't indicate anything about the `teardown` not being run theory.  I think currently the dataframes we append are either always the same or decreasing in timestamp.

File: python/benchmarks/basic_functions.py
Comment (grusev) [reply python/benchmarks/basic_functions.py:352]: pop(0) changed. Also updated the header of PR with trace of failure

Comment (grusev) [reply python/benchmarks/basic_functions.py]: fixed

File: python/benchmarks/basic_functions.py
Comment (grusev) [reply python/benchmarks/basic_functions.py:269]: fixed

File: python/benchmarks/finalize_staged_data.py
Comment (grusev) [reply python/benchmarks/finalize_staged_data.py:78]: This is in order to see results for benchmarks. They will fail because of guarded code uses something not avail in python 3.6. We should have it until workflow is upgraded to use it instead of 3.6

**Comment by grusev** [comment]
ASV time testing in more details with examples for ArcticDB": https://github.com/man-group/ArcticDB/wiki/Running-ASV-Benchmarks

**Comment by grusev** [comment]
Merging this one to https://github.com/man-group/ArcticDB/pull/2065

**Pull request closed without merging**
